src/js/references/types.ts

```diff
@@ -81,56 +82,56 @@ export type ReferenceNested = {
     name: string;
 };
 
-export class Reference {
-    clonedFrom: ReferenceClonedFrom | null;
-    contributors: ReferenceContributor[];
+export type ReferenceInstalled = {
+    body: string;
+    created_at: Date;
+    filename: string;
+    html_url: string;
+    id: number;
+    name: string;
+    newer: boolean;
+    published_at: Date;
+    ready: boolean;
+    size: number;
+    user: UserNested;
+};
+
+export type ReferenceBuild = {
     createdAt: Date;
-    dataType: ReferenceDataType;
-    description: string;
-    groups: ReferenceGroup[];
     id: string;
-    importedFrom: ReferenceImportedFrom | null;
-    installed: boolean;
-    internalControl: string | null;
-    latestBuild: ReferenceLatestBuild;
-    name: string;
+    user: UserNested;
+    version: number;
+    has_json: boolean;
+};
+
+export type ReferenceMinimal = ReferenceNested & {
+    cloned_from: ReferenceClonedFrom | null;
+    created_at: string;
+    imported_from: File | null;
+    installed: ReferenceInstalled;
+    internal_control: string | null;
+    latest_build: ReferenceBuild;
     organism: string;
-    otuCount: number;
-    release: ReferenceRelease | null;
-    remotesFrom: ReferenceRemotesFrom | null;
-    restrictSourceTypes: boolean;
-    sourceTypes: string[];
-    targets: ReferenceTarget[];
-    task: Task | null;
-    unbuiltChangeCount: number;
-    updating: boolean;
+    otu_count: number;
+    release: ReferenceRelease;
+    remotes_from: ReferenceRemotesFrom | null;
+    task: Task;
+    unbuilt_change_count: number;
+    updating: boolean | null;
     user: UserNested;
-    users: ReferenceUser[];
-
-    constructor(data) {
-        this.clonedFrom = data.cloned_from;
-        this.contributors = data.contributors;
-        this.createdAt = new Date(data.created_at);
```

Comment:
Make sure that the components that made use of this can handle the change. This class was converting python `snake case` variables to more java-script idiomatic `pascal case` naming.